hooks/source_links.py

```
@@ -64,11 +81,18 @@ def _rewrite(target, src_dir, repo_url):
     repo_path = _LEADING_UP.sub('', resolved)
-    kind = 'tree' if path.endswith('/') else 'blob'
-    url = '{}/{}/{}/{}'.format(repo_url, kind, _ref(), repo_path)
+    published = PUBLISHED.get(repo_path)
+    if published is not None:
+        # Relative to the linking page, so MkDocs resolves and validates it the
+        # way it would any internal link -- including the fragment, which must
+        # name a heading the published page actually has.
+        target = posixpath.relpath(published, src_dir) if src_dir else published
+    else:
+        kind = 'tree' if path.endswith('/') else 'blob'
+        target = '{}/{}/{}/{}'.format(repo_url, kind, _ref(), repo_path)
```

Ruff (0.16.1)

[warning] Use f-string instead of `format` call

Convert to f-string

(UP032)